src/effects/presets/effectchainpresetmanager.cpp

```diff
@@ -407,6 +407,35 @@ void EffectChainPresetManager::savePreset(EffectChainPresetPointer pPreset) {
     savePresetXml(pPreset);
 }
 
+void EffectChainPresetManager::updatePreset(EffectChainPointer pChainSlot) {
+    auto pPreset = EffectChainPresetPointer::create(pChainSlot.data());
+    updatePreset(pPreset);
+}
+
+void EffectChainPresetManager::updatePreset(EffectChainPresetPointer pPreset) {
+    const QString name = pPreset->name();
+    if (name.isEmpty()) {
+        return;
+    }
+
+    auto existing = m_effectChainPresets.find(name);
```

Comment:
I do not understand why you look up the EffectChainPresetPointer by name, when we already have the pointer. 
Please add a source code comment that explains the code.

Reply:
ah, I thought I could call reset() on the shared pointer and have it update both lists, but that does not work.